magento/pwa-studio
# [PWA-2174] Dynamic Blocks GraphQL returns non-qualified blocks

## Description

Refetch Dynamic Blocks when specific Cart conditions are updated.
Show relevant Dynamic Blocks matching Catalog Price Rules if Product UID is provided.
This PR requires the backend work from https://github.com/magento-commerce/magento2-pwa-commerce/pull/3.

## Related Issue







Closes #3418.

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature

Install PWA Commerce meta package with PR https://github.com/magento-commerce/magento2-pwa-commerce/pull/3.
1. Create new Cart Rule
2. Link Dynamic Block to rule
3. Add Dynamic Block to Category description via Page Build component
4. On frontend the Dynamic Block should only be shown when Cart Rule applies

Make sure to test possible Cart conditions:
- Subtotal (Excl. Tax)
- Subtotal (Incl. Tax)
- Subtotal
- Total Items Quantity
- Total Weight
- Payment Method
- Shipping Method
- Shipping Postcode
- Shipping Region
- Shipping State/Province
- Shipping Country

Test Catalog Price Rules on Simple Product Page.
1. Create new Catalog Price Rule
2. Link Dynamic Block to rule
3. Add Dynamic Block to a **Simple Product** description via Page Build component
4. On frontend the Dynamic Block should only be shown when Catalog Price Rule applies

#### Any ad-hoc/edge case scenarios that need to be considered?

- Test Cart conditions update on page where Dynamic Block is present
- Test Cart conditions update on page where Dynamic Block is not present

## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
